host/traceh.c

```diff
@@ -224,7 +315,70 @@ void oe_log_message(bool is_enclave, oe_log_level_t level, const char* message)
                 _log_creation_failed_before = true;
                 return;
             }
-            _write_message_to_stream(log_file, is_enclave, level, message);
+
+            if (!_custom_log_format)
+            {
+                _write_message_to_stream(
+                    log_file, is_enclave, time, usecs, level, message);
+            }
+            else
+            {
+                char* message_dup = strdup(message);
+                char* message_cursor = NULL;
+                char* log_msg = strtok_r(message_dup, "[", &message_cursor);
+                char* file_name = strtok_r(NULL, ":", &message_cursor);
+                char* function = strtok_r(NULL, ":", &message_cursor);
+                char* line_number = strtok_r(NULL, "]", &message_cursor);
+
+                if (!log_msg || !file_name || !function || !line_number)
+                {
+                    _write_message_to_stream(
+                        log_file,
+                        is_enclave,
+                        time,
+                        usecs,
+                        level,
+                        "Failed to apply custom formatter to message\n");
+                }
+                else
+                {
+                    if (_json_escape)
+                    {
+                        size_t msg_size = strlen(log_msg);
+                        char log_msg_escaped[2 * msg_size];
```

Comment:
Since unprintable values that aren't special cased effectively expand to 6 bytes, I think this may be too optimistic.